server/server/common/queries.py

```diff
@@ -1001,3 +1001,8 @@ class PWA:
     LIMIT 1
     RETURN {@uid: doc.filepath}
 '''
+
+SUTTA_PARENT = '''
+FOR parent_doc IN @level INBOUND DOCUMENT('super_nav_details', @uid) super_nav_details_edges
+    RETURN parent_doc.uid
+'''
```

Comment:
this query may contain the logic you described through python loops

Reply:
Yes, it can be executed up to ten times :smile: